src/Mvc/Mvc.Core/src/ControllerBase.cs

```diff
@@ -1870,7 +1870,10 @@ public virtual ConflictObjectResult Conflict([ActionResultObjectValue] ModelStat
                 detail: detail,
                 instance: instance);
 
-            return new ObjectResult(problemDetails);
+            return new ObjectResult(problemDetails)
+            {
+                StatusCode = statusCode ?? 500,
```

Comment:
I think you should get `statusCode` from `problemDetails`, since it's possible that the `statusCode` changes in `ProblemDetailsFactory`    
```suggestion
                StatusCode = problemDetails.Status ?? 500,
```

Reply:
Now that's the issue: status or statuscode should **not** be in the serialized result of the problem detail, otherwise we're not following the RFC. This needs a refactor IMO. (See my other comment on this PR).

Reply:
Sorry for the confusion. After a second look, there indeed is a status field in the json and xml output. I'll fix this PR.